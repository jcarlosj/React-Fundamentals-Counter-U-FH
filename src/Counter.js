import React, { useState } from 'react';
import PropTypes from 'prop-types'

/** Functional Component */
const Counter = ({ value }) => {

    const [ counter, setCounter ] = useState( 0 );

    const handleAdd = ( event ) => {
        setCounter( counter + 1 );      //  Otra Forma: setCounter( counter => counter + 1 ); 
    }

    return ( 
        <> 
            <h1>Counter</h1>
            <h2>{ counter }</h2>

            <button
                onClick={ handleAdd }
            >+ 1</button>
        </>
    );
}
 
Counter.propTypes = {
    value: PropTypes.number
}

export default Counter;

