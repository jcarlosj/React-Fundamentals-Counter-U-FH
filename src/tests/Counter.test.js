import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import Counter from '../Counter';

describe( '<Counter />', () => {
    
    describe( 'Enzime:', () => {
        
        test( 'should deploy the component correctly', () => {
           
            const wrapper = shallow( <Counter /> );

            // console.log( wrapper.debug() );

            expect( wrapper ).toMatchSnapshot();

        });

        test( 'should display the default value of 100 (as string)', () => {
            
            const 
                defaultValue = 100,
                wrapper = shallow( <Counter value={ defaultValue } /> ),
                h2Value = wrapper.find( 'h2' ).text().trim();       //  Encuentra un elemento h2 en el Componente

            // console.log( wrapper.debug() );

            expect( h2Value ).toBe( defaultValue.toString() );

        });
        

    });
    
});
