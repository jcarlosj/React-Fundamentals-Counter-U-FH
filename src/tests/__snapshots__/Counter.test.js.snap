// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Enzime: should deploy the component correctly 1`] = `
<Fragment>
  <h1>
    Counter
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    + 1
  </button>
  <button
    onClick={[Function]}
  >
    Reset 
  </button>
  <button
    onClick={[Function]}
  >
    - 1
  </button>
</Fragment>
`;
